src/main/java/net/snowflake/ingest/streaming/internal/AbstractRowBuffer.java

```diff
@@ -653,6 +677,42 @@ private void checkBatchSizeRecommendedMaximum(float batchSizeInBytes) {
     }
   }
 
+  /**
+   * We verify some offset expect behavior and report to SF if there is a mismatch. Note that there
+   * are false positives because the input could give us a batch with offset gaps. For example in
+   * Kafka Connector, we could have gaps if some of the offsets are filtered out by SMT.
+   */
+  private void checkOffsetMismatch(
+      String prevEndOffset, String curStartOffset, String curEndOffset, int rowCount) {
+    if (telemetryService != null && curStartOffset != null) {
+      boolean reportMismatch = false;
+      try {
+        long curStart = Long.parseLong(curStartOffset);
+        long curEnd = Long.parseLong(curEndOffset);
+
+        // We verify that the end_offset - start_offset + 1 = row_count
+        if (curEnd - curStart + 1 != rowCount) {
+          reportMismatch = true;
+        }
+
+        // We verify that start_offset_of_current_batch = end_offset_of_previous_batch+1
+        if (prevEndOffset != null) {
+          long prevEnd = Long.parseLong(prevEndOffset);
+          if (curStart != prevEnd + 1) {
+            reportMismatch = true;
```

Comment:
This should really be renamed `mismatchDetected` because technically reporting a mismatch is determined by having telemetry and/or logging enabled